python/lzreposync/src/lzreposync/primary_parser.py

```diff
@@ -50,61 +93,219 @@ def get_text(node):
+                        "obsoleteversion",
+                        "obsoleteflags",
+                        "conflictname",
+                        "conflictversion",
+                        "conflictflags",
+                        1159,
+                        1160,
+                        1161,
+                        1156,
+                        1157,
+                        1158,
+                        5052,
+                        5053,
+                        5054,
+                        5055,
+                        5056,
+                        5057,
+                        5049,
+                        5050,
+                        5051,
+                        5046,
+                        5047,
+                        5048,
+                    ]
+                    for dep in possibly_missing_dependencies:
+                        if not self.current_hdr.get(dep):
+                            self.current_hdr[dep] = []
+
+                    header_tags = [
+                        rpm.RPMTAG_DSAHEADER,
+                        rpm.RPMTAG_RSAHEADER,
+                        rpm.RPMTAG_SIGGPG,
+                        rpm.RPMTAG_SIGPGP,
+                        rpm.RPMTAG_FILEDIGESTALGO,
+                    ]
+                    for ht in header_tags:
+                        self.current_hdr[ht] = None
+
+                    # Parsing package's metadata
+                    for child_node in node.childNodes:
+                        if child_node.nodeType == child_node.ELEMENT_NODE:
+                            self.set_element_node(child_node)
+
+                    is_source = is_source_package(node)
+                    package_header = RPM_Header(self.current_hdr, is_source=is_source)
+                    self.current_package["header"] = package_header
+
+                    yield self.current_package
 
     def set_checksum_node(self, node):
         """
         Parse the given "checksum" node and the result Checksum object to the current package
         """
-        if not self.currentPackage:
+        if not isinstance(self.current_package, dict):
             print("Error: No package being parsed!")
             raise ValueError("No package being parsed")
 
-        checksum = Checksum()
-        # TODO: Surround with try except
-        checksum["type"] = node.attributes["type"].value
-        checksum["value"] = node.firstChild.nodeValue
-        self.currentPackage["checksum"] = checksum
+        self.current_package["checksum"] = get_text(node)
+        self.current_package["checksum_type"] = node.attributes["type"].value
 
     def set_attribute_element_node(self, node):
         """
         Parse the given attribute element node and add its information to the currentPackage.
         node: self-closing element. Eg: <version epoch="0" ver="1.22.0" rel="lp155.3.4.1"/>
         """
-        if not self.currentPackage:
+        # pylint: disable-next=unidiomatic-typecheck
```

Comment:
We can drop the pylint disable comment now:
```suggestion
        # pylint: disable-next=unidiomatic-typecheck
```